src/linesearches_compat.jl

```diff
@@ -0,0 +1,65 @@
+
+"""
+    LineSearchesStepsize <: Stepsize
+
+Wrapper for LineSearches.jl.
+"""
+struct LineSearchesStepsize{
+    TLS,TRM<:AbstractRetractionMethod,TVTM<:AbstractVectorTransportMethod
+} <: Stepsize
+    linesearch::TLS
+    retraction_method::TRM
+    vector_transport_method::TVTM
+end
+function LineSearchesStepsize(
+    M::AbstractManifold,
+    ls;
+    retraction_method=default_retraction_method(M),
+    vector_transport_method=default_vector_transport_method(M),
+)
+    return LineSearchesStepsize{
+        typeof(ls),typeof(retraction_method),typeof(vector_transport_method)
+    }(
+        ls, retraction_method, vector_transport_method
+    )
+end
+
+function (cs::LineSearchesStepsize)(
+    mp::AbstractManoptProblem,
+    s::AbstractManoptSolverState,
+    i::Int,
+    η=-get_gradient(mp, get_iterate(s));
+    fp=get_cost(mp, get_iterate(s)),
+    kwargs...,
+)
+    M = get_manifold(mp)
+    p = get_iterate(s)
+    p_tmp = copy(M, p)
+    X_tmp = copy(M, p, η)
+    Y_tmp = copy(M, p, η)
+    f = get_cost_function(get_objective(mp))
+    function ϕ(α)
```

Comment:
These three functions could also be a functor/struct as a field in the wrapper struct? (Hope this idea is not sketched too short but I am thinking similar to AugmentedLagrangianCost & such).

Reply:
I will try to modify it so that temporaries are only allocated once per optimization, so this is just a first sketch anyway.

Reply:
Sure, only do that if this works. And I think the same idea as with structs is the way to go afterwards – but since I learned that approach with functors from you, I am sure you know what to do.

Reply:
Hm, this is complicated. I can't easily allocate these buffers at the start of optimization. The easiest way would be by adding a function that is called during state creation that asks line search what kind of buffer it needs, puts it in the state, and allows for access by line searches using another new function. Does that sound reasonable?

I've considered converting these three functions to functors and it seems that it will only make the code unnecessarily convoluted in this case. Functors are great for encapsulating what would otherwise be global state, but accessing local state (here: variables in the outer function) rarely causes performance issues.

Reply:
I think it would avoid the allocations for X_tmp and Y_tmp though because that would be fields in phi_struct (or in the line search wrapper)

Reply:
`StepsizeStorage` should allocate these temporaries but I forgot to redo this part.

Reply:
Do we still wanna store the cantons/functionals in the wrapper struct? Or do you think that does not have much of a advantage allocation wise.

Reply:
It has no advantage whatsoever at the moment as far as I can tell. The struct-based code would even have to be fairly convoluted to not be *worse* than the current solution allocation-wise.

Reply:
ok, then lets leave this as is :)